internal/webserver/handler-api.go

```diff
@@ -323,7 +323,7 @@ func (h *handler) apiInsertBookmark(w http.ResponseWriter, r *http.Request, ps h
 	if !payload.Async {
 		book, err = downloadBookmarkContent(book, h.DataDir, r)
 		if err != nil {
-			log.Printf("error downloading boorkmark: %s", err)
+			panic(fmt.Errorf("failed to save bookmark: %v", err))
```

Comment:
I don't like this approach, mostly because:
1) I don't like the usage of panic/recover, it makes following errors complicated for contributors.
2) We don't want to stop here if the download fails, we just need to report the error and save the bookmark anyway, since the user can trigger a manual download on its own.

Reply:
Yes, I agree. #513 is a better solution.